### DISCUSSION TITLE
How much to show "Keep a record" Pattern?

### DISCUSSION THREAD

**Discussion by jeana-adhoc** [uswds/uswds#5940] [Q&A]
The USWDS guidance for the "Keep a record" pattern includes

> Providing the user with a record of the answers provided, questions skipped, and the URL and date stamp of form submission can reduce fear, increase confidence, and improve the user’s trust in your program.

My question is about "Questions skipped". In complex form flows, there may be conditional flows that the user never encountered, I'm not sure we should reveal those in the pattern? We should definitely show the question that enabled a flow to be skipped. But, I just wanted to clarify if "questions skipped" also included conditionals.

Also - for questions with checkboxes or radio selections, should the record also show items that were *not* selected?  

Thanks,
Jeana

**Comment by mejiaj** [comment]
@jaclinec are you aware of any existing UX patterns that address this?

**Comment by jaclinec** [comment]
@jeana-adhoc I don't think conditionals should be included in records of answers provided, because I think that the risk of causing confusion to the user ("I didn't even see these questions, what's going on here?!") might outweigh any potential benefits. As with most things, I'd try to think through the possible use cases of why a participant might want to access conditional questions not asked of them and make a judgement call on whether to include it in the record.

As for the multiple choice questions, that's a bit of a tougher call. I see the benefit in providing all the options. It could be useful for users to see what _wasn't_ selected in case they mistakenly didn't choose something that they should have/meant to select.

**Comment by jeana-adhoc** [comment]
This all makes sense to me and where I was leaning myself. 

But I was confused by the [guidance that said](https://designsystem.digital.gov/patterns/complete-a-complex-form/keep-a-record/) "Provide a user with a record of ... questions skipped" and thought we might need to nuance that a bit.

**Reply by jeana-adhoc** [reply]
Then again, maybe "questions skipped" are just "not required" questions.... questions the user was presented with, but did not answer...

**Reply by jaclinec** [reply]
Yes  I think we are referring to non required questions that the user made a decision to not answer. I agree that there's room to clarify what we mean in the guidance - I created [#2703](https://github.com/uswds/uswds-site/issues/2703) to address this.

**Comment by ryparker** [comment]
“Questions skipped” means optional questions the user was shown but chose not to answer. It does not mean every question hidden by conditional logic. That distinction was confirmed in this thread and in [the documentation correction, #2703](https://github.com/uswds/uswds-site/issues/2703).

For checkboxes and radio buttons, make the submitted answers unambiguous. Showing unselected choices can help someone review a possible omission, but the pattern does not require displaying every unselected option in every form. Include that extra context when it helps the user understand what they submitted, and clearly distinguish an unanswered optional question from an explicit answer such as “None.”
